src/components/Instructions.component.js

```diff
@@ -1,21 +1,32 @@
 import React, { Component } from 'react'
-// TODO: uncomment this for Part 1
-// import PropTypes from 'prop-types'
+import PropTypes from 'prop-types'
 import './../styles/instructions.css'
 
 class Instructions extends Component {
   render() {
     return (
       <div className="instructions">
-        Follow the instructions on the README to get started!
-        <img
-          className="instructions__logo"
-          src="https://uiuc.hack4impact.org/img/colored-logo.png"
-          alt="h4i logo"
-        />
+        Follow the instructions on the README to get started
+        {this.props.shouldDisplayImage && (
+          <img
+            className="instructions__logo"
+            src="https://uiuc.hack4impact.org/img/colored-logo.png"
+            alt="h4i logo"
+          />
+        )}
+        <ul>
+          {this.props.list
+            .map(item => item.toUpperCase())
+            .filter(item => item.length < 3)
+            .map(item => <li>{item}</li>)}
```

Comment:
You can minimize the number of map statements using. Map statements are faster and cleaner than for loops, but every time you use one it creates a new copy of the list, so combining them helps :) 
```
.map(item => <li>{item.toUpperCase(}</li>)}
```

Reply:
that is _technically_ true, but in these kind of cases, i would say stick with cleaner and easier to follow code. these are very small lists so the performance issues are negligible. also, there are compiler optimizations that combine `map`s and `filter`s and other chainable array functions into a single pass, even if you write your code in separate statements.